lib/src/widgets/adw/raw_target.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:flutter/material.dart';
+
+/// Set of status that a [RawAdwTarget] widget can be at any given time.
+enum TargetStatus { enabled, active, hover, tapDown }
+
+typedef TargetColorBuilder = Color? Function(BuildContext, TargetStatus);
+
+typedef TargetWidgetBuilder = Widget Function(BuildContext, TargetStatus);
+
+/// Basic widget that can be used to create other more complex and customized
+/// buttons and other set of 'target' views.
+///
+/// It offers the ability to reconstruct sets of properties depending on the
+/// current state of the view.
+class RawAdwTarget extends StatefulWidget {
+  const RawAdwTarget({
```

Comment:
Do you see this class being used for more than just buttons? Could it/would it be used for stuck like "AdwComboBox"? if no I would rename the class to something more "understandable" such as `AdwBaseButton`.

I would also propose to clarify the documentation a little bit. The code is very clear, easy to understand, and well made but the documentation sounded vague to me. What about something like:

```dart
/// RawAdwTarget is a widget used as a base to build Adwaita widgets
/// with pressed and hover properties.
///
/// RawAdwTarget widgets can rebuild their background through
/// the [this.backgroundColorBuilder] callback.
```

Reply:
I do think that this widget should be the base of all components that are 'clickable' by the user. This is useful to handle changes when a user hovers or clicks on the view.

So yes, This widget should support a widget like `AdwComboBox`. Maybe in the process of using this new widget inside the combo box we find some aspect that could be improved or changes.

I'll update the widget's documentation, since the one you've proposed looks better than the current one :)

Reply:
Sounds good to me then :).
Let's keep in mind the migration of Combobox for later then!